drivers/ng_at86rf2xx/ng_at86rf2xx.c

```diff
@@ -233,9 +233,15 @@ void ng_at86rf2xx_tx_exec(ng_at86rf2xx_t *dev)
 
 size_t ng_at86rf2xx_rx_len(ng_at86rf2xx_t *dev)
 {
-    uint8_t res;
-    ng_at86rf2xx_fb_read(dev, &res, 1);
-    return (size_t)(res - 2);           /* extract the PHR and LQI field */
+    uint8_t phr;
+    ng_at86rf2xx_fb_read(dev, &phr, 1);
+
+    /* Mask MSB for 802.15.4 compliance, see
+     * section 37.1.1.2 of SAM R21 datahseet
```

Comment:
could you please reference the radio datasheets (-> it is section 8.1.1.2 for all four supported devices...)